dxr/plugins/python/indexers.py

```diff
@@ -153,30 +227,35 @@ def __init__(self, path, contents, plugin_name, tree, python_path,
         self.derived_classes = derived_classes
         self.names = names
 
+        self.module_name = path_to_module(self.python_path, self.path)
+
     def is_interesting(self):
         return is_interesting(self.path)
 
     def needles_by_line(self):
+        self.analyze_tokens()
+
+        visitor = IndexingNodeVisitor(self)
+        syntax_tree = ast.parse(self.contents.encode('utf-8'))
```

Comment:
Okay, here's the skinny: the ast module works in terms of UTF-8 offsets. From the ast docs…

> the col_offset is the UTF-8 byte offset of the first token that generated the node

It'll crash if fed a unicode with certain explicit encoding declarations up top. If we pass it a utf-8 bytestring, it doesn't try parsing the declaration. Se we should keep this as is, and perhaps we should change the tokenize call in some way. We'll also need to convert back from utf-8 offsets to unicode ones to conform to DXR's API, but that can wait if we ticket it. That shouldn't be too labor-intense if we make use of FileToSkim._line_offsets().